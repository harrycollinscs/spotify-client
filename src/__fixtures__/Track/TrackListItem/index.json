{
  "track": {
    "album": {
      "album_type": "ALBUM",
      "artists": [
        {
          "external_urls": {
            "spotify": "https://open.spotify.com/artist/66l6SXrWHLjXFHfv202kdA"
          },
          "href": "https://api.spotify.com/v1/artists/66l6SXrWHLjXFHfv202kdA",
          "id": "66l6SXrWHLjXFHfv202kdA",
          "name": "Pastor T.L. Barrett and the Youth for Christ Choir",
          "type": "artist",
          "uri": "spotify:artist:66l6SXrWHLjXFHfv202kdA"
        }
      ],
      "available_markets": [
        "AD",
        "AE",
        "AR",
        "AT",
        "AU",
        "BE",
        "BG",
        "BH",
        "BO",
        "BR",
        "CA",
        "CH",
        "CL",
        "CO",
        "CR",
        "CY",
        "CZ",
        "DE",
        "DK",
        "DO",
        "DZ",
        "EC",
        "EE",
        "EG",
        "ES",
        "FI",
        "FR",
        "GB",
        "GR",
        "GT",
        "HK",
        "HN",
        "HU",
        "ID",
        "IE",
        "IL",
        "IN",
        "IS",
        "IT",
        "JO",
        "JP",
        "KW",
        "LB",
        "LI",
        "LT",
        "LU",
        "LV",
        "MA",
        "MC",
        "MT",
        "MX",
        "MY",
        "NI",
        "NL",
        "NO",
        "NZ",
        "OM",
        "PA",
        "PE",
        "PH",
        "PL",
        "PS",
        "PT",
        "PY",
        "QA",
        "RO",
        "SA",
        "SE",
        "SG",
        "SK",
        "SV",
        "TH",
        "TN",
        "TR",
        "TW",
        "US",
        "UY",
        "VN",
        "ZA"
      ],
      "external_urls": {
        "spotify": "https://open.spotify.com/album/7bsJSHsfKm5dXbLyY6zba4"
      },
      "href": "https://api.spotify.com/v1/albums/7bsJSHsfKm5dXbLyY6zba4",
      "id": "7bsJSHsfKm5dXbLyY6zba4",
      "images": [
        {
          "height": 640,
          "url": "https://i.scdn.co/image/ab67616d0000b273902c717af274bfcfb232721f",
          "width": 640
        },
        {
          "height": 300,
          "url": "https://i.scdn.co/image/ab67616d00001e02902c717af274bfcfb232721f",
          "width": 300
        },
        {
          "height": 64,
          "url": "https://i.scdn.co/image/ab67616d00004851902c717af274bfcfb232721f",
          "width": 64
        }
      ],
      "name": "Like a Ship (Without a Sail)",
      "release_date": "1971-01-01",
      "release_date_precision": "day",
      "total_tracks": 8,
      "type": "album",
      "uri": "spotify:album:7bsJSHsfKm5dXbLyY6zba4"
    },
    "artists": [
      {
        "external_urls": {
          "spotify": "https://open.spotify.com/artist/66l6SXrWHLjXFHfv202kdA"
        },
        "href": "https://api.spotify.com/v1/artists/66l6SXrWHLjXFHfv202kdA",
        "id": "66l6SXrWHLjXFHfv202kdA",
        "name": "Pastor T.L. Barrett and the Youth for Christ Choir",
        "type": "artist",
        "uri": "spotify:artist:66l6SXrWHLjXFHfv202kdA"
      }
    ],
    "available_markets": [
      "AD",
      "AE",
      "AR",
      "AT",
      "AU",
      "BE",
      "BG",
      "BH",
      "BO",
      "BR",
      "CA",
      "CH",
      "CL",
      "CO",
      "CR",
      "CY",
      "CZ",
      "DE",
      "DK",
      "DO",
      "DZ",
      "EC",
      "EE",
      "EG",
      "ES",
      "FI",
      "FR",
      "GB",
      "GR",
      "GT",
      "HK",
      "HN",
      "HU",
      "ID",
      "IE",
      "IL",
      "IN",
      "IS",
      "IT",
      "JO",
      "JP",
      "KW",
      "LB",
      "LI",
      "LT",
      "LU",
      "LV",
      "MA",
      "MC",
      "MT",
      "MX",
      "MY",
      "NI",
      "NL",
      "NO",
      "NZ",
      "OM",
      "PA",
      "PE",
      "PH",
      "PL",
      "PS",
      "PT",
      "PY",
      "QA",
      "RO",
      "SA",
      "SE",
      "SG",
      "SK",
      "SV",
      "TH",
      "TN",
      "TR",
      "TW",
      "US",
      "UY",
      "VN",
      "ZA"
    ],
    "disc_number": 1,
    "duration_ms": 390680,
    "explicit": false,
    "external_ids": {
      "isrc": "USCW61018005"
    },
    "external_urls": {
      "spotify": "https://open.spotify.com/track/13DgFRTlpButNOdGQaAzbK"
    },
    "href": "https://api.spotify.com/v1/tracks/13DgFRTlpButNOdGQaAzbK",
    "id": "13DgFRTlpButNOdGQaAzbK",
    "is_local": false,
    "name": "Nobody Knows",
    "popularity": 44,
    "preview_url": "https://p.scdn.co/mp3-preview/ffcb5b4b9dd5407450e38fc3809e6e2c5fe964ca?cid=a4f47b9ec311413bae51f77838af3e3d",
    "track_number": 5,
    "type": "track",
    "uri": "spotify:track:13DgFRTlpButNOdGQaAzbK"
  },
  "offset": 17,
  "playlistId": null,
  "albumId": null
}
