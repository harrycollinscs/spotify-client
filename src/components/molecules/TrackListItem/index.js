import React from "react";
import styled from "styled-components";
import { faPlay } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import axios from "axios";
import { me_endpoint } from "../../../config";
import store from "../../../store";
import { playTrack } from "../../../api/songs";

const Item = styled.div`
  display: flex;
  flex-direction: row;
  height: 50px;
  margin: 5px;
`;

const Song = styled.p`
  padding: 0px;
  margin: 0px;
`;

const Artists = styled.p`
  padding: 0px;
  margin: 0px;
  color: grey;
`;

const Divider = styled.div`
  background-color: #eee;
  width: 100%;
  height: 1px;
`;

const ContentContainer = styled.div`
  margin-left: 10px;
`;

const StyledImage = styled.img`
  height: 50px;
  width: auto;
`;

const StyledImageOverlay = styled.div`
  height: 50px;
  width: 50px;
  background-color: black;
  opacity: 0;
  position: absolute;
  bottom: 0;
  transition: .5s ease;
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;

  &:hover {
    background-color: black;
    opacity: 0.75;
  }
`;

const IconStyle = {
  color: "#fff",
  margin: "5px 5px 5px 5px",
  height: "20px",
  width: "20px",
};

const getTrack = (track) => {
  return track.track ? track.track : track;
};

const TrackListItem = ({ track, offset, playlistId = null, albumId = null }) => {
  const { name, artists, album } = getTrack(track);

  console.log(offset);

  const playPayload = {
    type: playlistId ? 'playlist' : 'album',
    id: playlistId ? playlistId : albumId,
    offset: offset
  };

  const allArtists = artists
    .map((artist) => {
      return artist.name;
    })
    .join(", ");

  return (
    <div>
      <Item>

      <div style={{position: 'relative'}}>
        <StyledImage src={album.images[0].url} />
        <StyledImageOverlay onClick={() => playTrack(playPayload)} >
          <FontAwesomeIcon icon={faPlay} style={IconStyle} />
        </StyledImageOverlay>
      </div>

        <ContentContainer>
          <Song>{name}</Song>
          <Artists>{allArtists}</Artists>
        </ContentContainer>
      </Item>
      <Divider />
    </div>
  );
};

export default TrackListItem;
