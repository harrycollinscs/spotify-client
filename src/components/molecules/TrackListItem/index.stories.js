import React from "react";
import TrackListItem from "./index";
import TrackListItemData from "../../../__fixtures__/Track/TrackListItem/index.json";

export default {
  title: "Molecules/TrackListItem",
  component: TrackListItem,
};

const Template = (args) => <TrackListItem {...args} />;

export const Default = Template.bind({});
Default.args = {
  ...TrackListItemData
};
