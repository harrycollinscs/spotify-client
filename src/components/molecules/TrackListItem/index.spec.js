import React from "react";
import TrackListItem from "./index";
import { shallow } from "enzyme";
import TrackData from "../../../__fixtures__/Track/TrackListItem";

describe("Molecules/TrackListItem", () => {
  const component = <TrackListItem {...TrackData} />;

  it("Should match snapshot", () => {
    const render = shallow(component);
    expect(render).toMatchSnapshot();
  });
});
