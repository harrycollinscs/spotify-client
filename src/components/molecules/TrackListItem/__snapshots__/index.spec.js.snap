// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/TrackListItem Should match snapshot 1`] = `
<div>
  <styled.div>
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <styled.img
        src="https://i.scdn.co/image/ab67616d0000b273902c717af274bfcfb232721f"
      />
      <styled.div
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f04b",
                "M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z",
              ],
              "iconName": "play",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          style={
            Object {
              "color": "#fff",
              "height": "20px",
              "margin": "5px 5px 5px 5px",
              "width": "20px",
            }
          }
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </styled.div>
    </div>
    <styled.div>
      <styled.p>
        Nobody Knows
      </styled.p>
      <styled.p>
        Pastor T.L. Barrett and the Youth for Christ Choir
      </styled.p>
    </styled.div>
  </styled.div>
  <styled.div />
</div>
`;
